### PR TITLE
Add the array/arraysegment->Span/Memory overloads to MemoryExtensions

### PR DESCRIPTION
**Pull request by AtsushiKan**
Part of https://github.com/dotnet/corefx/issues/26894

### CODE REVIEW

**Comment by ahsonkhan** [comment]
Need to rebase on top the changes from the mirror: https://github.com/dotnet/corefx/pull/27586

**Comment by AtsushiKan** [comment]
@dotnet-bot test Tizen armel Debug Build

File: src/System.Memory/tests/Memory/AsMemory.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            Assert.Throws<ArgumentOutOfRangeException>(() => a.AsMemory(start));
+        }
+
+        [Theory]
+        [InlineData(0, -1, 0)]
+        [InlineData(0, 1, 0)]
+        [InlineData(0, 0, -1)]
+        [InlineData(0, 0, 1)]
+        [InlineData(5, 6, 0)]
+        [InlineData(5, 3, 3)]
```
Comment (ahsonkhan) [inline src/System.Memory/tests/Memory/AsMemory.cs]: add a case where both start and sublength are negative.

File: src/System.Memory/tests/Span/AsSpan.cs [hunk 2]
```diff
@@ -95,5 +95,116 @@ public static void ZeroLengthArraySegmentAsSpan()
...
+        [InlineData(3, 0)]
+        [InlineData(3, 1)]
+        [InlineData(3, 2)]
+        [InlineData(3, 3)]
+        [InlineData(10, 0)]
+        [InlineData(10, 3)]
+        [InlineData(10, 10)]
+        public static void ArraySegmentAsSpanWithStart(int length, int start)
+        {
+            const int segmentOffset = 5;
```
Comment (ahsonkhan) [inline src/System.Memory/tests/Span/AsSpan.cs]: nit: const use PascalCasing

**Comment by ahsonkhan** [comment]
Looks good. Ty for getting this in :)

**Pull request merged**